### ISSUE TITLE
a couple comments on AlexNet/Keras notebook

### ISSUE THREAD

**Issue by jg10545** [bootsinthecity/AlexNet-with-Keras-Exercise#1]
Hey, the exercise looks great. A couple things in no particular order:

* your conda instructions should say `. activate TF` (the name of your new env), not `. activate py35`
* regarding "Unclear what the strides are in the paper"- you can infer from the diagram in the paper that they're all stride 1
* the AlexNet paper uses overlapping pooling layers- pooling 3 pixels with a stride of 2
* did you batch norm the convolutional layers? I can't tell, but I also don't use Keras much
* the 1000-neuron dense layer in AlexNet is the output layer (i.e. the logits). so your network is adding one layer on top of AlexNet.

**Comment by jg10545** [comment]
No trouble running your Keras notebook on a GPU. Training takes about 2ms/step.

It looks like sometimes the training gets stuck in local minima, and if it doesn't improve after a few dozen epochs you might as well restart. Otherwise I'm getting validation accuracies in the 54% range.

**Comment by lewaggs** [comment]
Thanks Joe! I'm just running on my MacBook, so it's going much slower (about 70ms/step)... but that's good to know. I just added batch norm in the viz notebook after mistakenly leaving it out and that helped improve performance drastically from what I was getting without, not surprisingly.

**Comment by bootsinthecity** [comment]
Joe -

I'm on the plane on the way to Seattle and finally had the chance to go
back and look at these comments in detail.


   - your conda instructions should say . activate TF (the name of your new
   env), not . activate py35
      - fixed
   - regarding "Unclear what the strides are in the paper"- you can infer
   from the diagram in the paper that they're all stride 1
      - fixed
   - the AlexNet paper uses overlapping pooling layers- pooling 3 pixels
   with a stride of 2
      - Unsure if I can implement this detail in keras - omitting for now
   - did you batch norm the convolutional layers? I can't tell, but I also
   don't use Keras much
      - added batch norm - you add it at the end
   - the 1000-neuron dense layer in AlexNet is the output layer (i.e. the
   logits). so your network is adding one layer on top of AlexNet.
      - I noticed that we needed an extra layer to map to the 17 classes in
      oxflower
      - I guess we could have omitted the layer with 1000 and replaced it
      with 17 instead?

I will push these to git as soon as I finish running 100 epochs. I will
then graph the outputs. I will tackle tensorflow next. This was - more than
anything - a confidence-building exercise for me and to clarify my
understanding of neural network structure at a bare minimum so that I don't
feel overwhelmed by it all. This helped a lot.

Thanks again for your help. I feel like I've made so much progress in the
last six months and it seems to come ever faster as of late.

Hope y'all have fun with your visits today!

~Amanda

**Comment by bootsinthecity** [comment]
Pushed. :)
